lib/game_stats.rb

```diff
@@ -1,4 +1,35 @@
 
 
 module GameStats
+
+    def count_of_games_by_season 
+        games_per_season = Hash.new(0)
+        @games.each do |game|
+            if games_per_season.has_key?(game.season_id)
+                games_per_season[game.season_id] += 1
+            else
+                games_per_season[game.season_id] = 1
+            end
+        end
+        games_per_season
+    end
+
+    def average_goals_per_game
+        average_of_games = []
+        @games.each do |game|
+            average_of_games << game.away_goals.to_f
+            average_of_games << game.home_goals.to_f
+        end
+        average_of_games = average_of_games.sum / average_of_games.size
```

Comment:
might be good to set this variable to a different name than the array that you created (I haven't run the code so I don't know if it would cause any problems). maybe set the array name to something like all_goals and then make average_of_gams the result of the division you do in this line

Reply:
Additionally (sorry, just noticed this). I believe you'd want to do average_of_games.sum / (average_of_games.size / 2). The array you loaded with all of the goal values has two elements per game (home goals AND away goals) so when you use the .size you're actually dividing by double the amount of actual games

Reply:
Oh yeah! That makes a lot of sense (that being home and away goals for a game), I totally overlooked that. Additionally, just for following good design I think I should change the name of average_of_games. It did not seem to effect the tests but I think it should definitely still be refactored.